app/android/src/uk/co/lutraconsulting/InputActivity.java

```diff
@@ -115,6 +122,41 @@ public void hideSplashScreen()
     keepSplashScreenVisible = false;
   }
 
+public void showPDF(String filePath) {
+  Log.d( TAG, "Expected file path: " + filePath );
+
+  File file = new File( filePath );
+
+  if ( !file.exists() ) {
+      Log.d( TAG, "File does not exist: " + filePath );
+      runOnUiThread(() -> Toast.makeText( getApplicationContext(), "File not available", Toast.LENGTH_SHORT).show() );
+      return;
+  } else {
+      Log.d( TAG, "File exists: " + filePath );
+  }
+
+  Intent showFileIntent = new Intent( Intent.ACTION_VIEW );
+  if ( showFileIntent.resolveActivity( getPackageManager() ) != null ) {
+
+      try {
+          Uri fileUri = FileProvider.getUriForFile( this, "uk.co.lutraconsulting.fileprovider", file );
+          Log.d( TAG, "File URI: " + fileUri.toString() );
+
+          showFileIntent.setData( fileUri );
+          showFileIntent.setFlags( Intent.FLAG_ACTIVITY_NO_HISTORY | Intent.FLAG_GRANT_READ_URI_PERMISSION );
+          showFileIntent.addFlags( Intent.FLAG_ACTIVITY_NEW_TASK );
```

Comment:
I believe you need to do all of these steps prior to calling `showFileIntent.resolveActivity`. The way you implement it now you tell Android that you want to open _something_ and ask it if it has an app for that. But it does not know what you are trying to open. 

You should first set all these `setData`,`setFlags` and only after that call the check to see if there is any app that can open what you want